File: src/main/java/com/kelsos/mbrc/data/ConnectionSettings.java
package com.kelsos.mbrc.data;

import org.codehaus.jackson.JsonNode;

public class ConnectionSettings {
    private String address;
    private String name;
    private int port;

    public ConnectionSettings(JsonNode node){
        this.address = node.path("address").asText();
        this.name = node.path("name").asText();
        this.port = node.path("port").asInt();
    }

    public ConnectionSettings(String address, String name, int port) {
        this.address = address;
        this.name = name;
        this.port = port;
    }

    public String getAddress() {
        return this.address;
    }

    public String getName() {
        return this.name;
    }

    public int getPort() {
        return this.port;
    }

    @Override public boolean equals(Object o) {
        boolean equality = false;

        if (o instanceof ConnectionSettings) {
            ConnectionSettings other = (ConnectionSettings)o;
            equality = other.getAddress().equals(address) &&
                    other.getName().equals(name) &&
                    other.getPort() == port;
        }
        return equality;
    }
}

File: app/src/main/java/com/kelsos/mbrc/data/ConnectionSettings.kt
package com.kelsos.mbrc.data

import com.kelsos.mbrc.data.db.RemoteDatabase
import com.raizlabs.android.dbflow.annotation.Column
import com.raizlabs.android.dbflow.annotation.ConflictAction
import com.raizlabs.android.dbflow.annotation.PrimaryKey
import com.raizlabs.android.dbflow.annotation.Table
import com.raizlabs.android.dbflow.annotation.Unique
import com.raizlabs.android.dbflow.annotation.UniqueGroup

@Table(
  name = "settings",
  database = RemoteDatabase::class,
  uniqueColumnGroups = [UniqueGroup(groupNumber = 1, uniqueConflict = ConflictAction.IGNORE)],
)
data class ConnectionSettings(
  @Column(name = "address")
  @Unique(unique = false, uniqueGroups = [1])
  var address: String? = null,
  @Unique(unique = false, uniqueGroups = [1])
  @Column(name = "port")
  var port: Int = 0,
  @Column(name = "name")
  var name: String? = null,
  @PrimaryKey(autoincrement = true)
  @Column(name = "id")
  var id: Long = 0,
)
